--- configure.ac.orig
+++ configure.ac
@@ -23,7 +23,9 @@ AC_SUBST(ACLOCAL_AMFLAGS, "-I M4")
 
 AC_LANG([C])
 
-AC_PROG_CC
+AC_PROG_CC_STDC
+AC_USE_SYSTEM_EXTENSIONS
+AC_SYS_LARGEFILE
 AM_PROG_CC_C_O
 AC_PROG_CXX
 AC_PROG_SED
--- src/file_io.c.orig
+++ src/file_io.c
@@ -564,6 +564,9 @@ psf_open_fd (PSF_FILE * pfile)
 				return - SFE_BAD_OPEN_MODE ;
 				break ;
 		} ;
+#ifdef O_CLOEXEC
+        oflag |= O_CLOEXEC;
+#endif
 
 	if (mode == 0)
 		fd = open (pfile->path.c, oflag) ;
--- Makefile.am.orig
+++ Makefile.am
@@ -1,5 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
+ACLOCAL_AMFLAGS = -I M4
 DISTCHECK_CONFIGURE_FLAGS = --enable-gcc-werror
 
 if BUILD_OCTAVE_MOD
